styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```diff
@@ -322,9 +329,12 @@ public void create(Environment environment) {
 
     warmUpCache(workflowCache, storage);
 
+    // TODO: hack to get around circular reference. Change OutputHandler.transitionInto() to
+    //       take StateManager as argument instead?
```

Comment:
Is this comment still valid? Moving the OutputHandler into the StateManager seems like a design choice in this PR rather than a workaround.

Reply:
The circular dependency is still there. The code is passing in the list of output handlers into the `QueuedStateManager` constructor, and then passing in the `QueuedStateManager` instance into the constructors of the output handlers.

As the comment says, I would suggest breaking this cycle by not passing in the `QueuedStateManager` instance into the output handler constructors. Instead have the state manager pass itself as an argument to `OutputHandler.transitionInto()`.

Reply:
Gotcha. Unfortunately passing StateManager as argument to `OutputHandler.transitionInto()` is also not great since `OutputHandler` interface should have no relations with `StateManager`. Indeed, some `OutputHandlers` don't interact with any `StateManager`.